### PR TITLE
Ticket #2158: Use biometric auth instead of IAL2

### LINKED ISSUE #2158: Update Login.gov config to use Biometric configuration
### Issue description

Login.gov is doing their biometric beta program and we want to participate. For new users, they should go through this new identity verification program, but old users should go through

### Acceptance criteria

- [ ] "All New Users Require Biometric Flow" configurations changes are made to our oidc setup, thus requiring new users to go through the biometric flow and previous users to go through current idv options
- [ ] update unit tests
- [ ] remove step_up_acr_value (and any references)

### Additional context

[slack thread](https://cisa-corp.slack.com/archives/C05BGB4L5NF/p1715248591837839) for reference

See the [documentation for Biometric configuration](file:///Users/broddrick/Downloads/Login.gov%20-%20Biometric%20Comparison%20Limited%20Pilot.pdf) and specifically page 4 where it mentions "All New Users Require Biometric Flow"

Notes for dev:
settings.py sets the OIDC_PROVIDERS values which is what controls our login.gov config for both identity sandbox and production. By looking at the documentation it looks like we can leave our ial1 logic alone and then update our logic for step logic. For ial2 we will remove the step up acr value which is no longer needed per their documentation and add vtr and vtm verification.

settings fields
image

ial2 setup where acr values need to change
image

### Links to other issues

_No response_

### PR DESCRIPTION
**Pull request by zandercymatics**
## Ticket

Resolves #2158

## Changes

- Replaces our IAL2 check with two variables that do biometric authentication: vtm and vtr
- Removed references to step_up_acr_value
- Minor refactors for test cases and linter

## Context for reviewers
This ticket implements biometric authentication in ooidc for login.gov. 

## Setup
To test this ticket you will need:
1. An account that does not need elevated authentication: i.e. one that is created with fixtures
2. An account that DOES need authenticated authentication: i.e. one that wasn't created by fixtures, or a brand new account. If you wish to create a new one, [use this service](https://www.guerrillamail.com/) if you cannot easily create a dummy email. 

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

**Comment by abroddrick** [comment]
@zandercymatics Don't merge this one until I confirm with login.gov that we are good to go into production with this. emailing them now

**Review by abroddrick — CHANGES_REQUESTED** [review]
Hey, I would expect a unit test testing the step up to ial2 (aka the biometric authentication), which did exist but you deleted. Instead of deleting the unit test I would edit the test to now fit with the new step up flow.

Side note, you don't need to call out that it uses biometrics, it's still a step authentication to "ial2" (even though our previous one, which we called ial2 , really wasn't ial2 yet)

**Review by abroddrick — APPROVED** [review]
Approving though I see Rachid also assigned himself this one. I tested and the code changes make sense based on the documentation.

File: src/djangooidc/views.py [hunk 1]
```diff
@@ -91,12 +91,21 @@ def login_callback(request):
...
-        if _requires_step_up_auth(userinfo):
-            # add acr_value to request.session
-            request.session["acr_value"] = CLIENT.get_step_up_acr_value()
-            return CLIENT.create_authn_request(request.session)
+        # if not satisfied, redirect user to login requiring biometric auth
+
+        # Tests for the presence of the vtm/vtr values in the userinfo object.
+        # If they are there, then we can set a flag in our session for tracking purposes.
+        needs_step_up_auth = _requires_step_up_auth(userinfo)
+        request.session["needs_step_up_auth"] = needs_step_up_auth
```
Comment (rachidatecs) [inline src/djangooidc/views.py:101]: Q: should this be moved into the if block?

**Review by rachidatecs — APPROVED** [review]
See Q

**Pull request merged**